src/base/dense/owl_base_dense_ndarray.ml

```diff
@@ -23,13 +23,78 @@ module S = struct
   include Owl_base_dense_ndarray_s
   include Operator
   module Scalar = Owl_base_maths
+
+  let eye = Owl_base_dense_matrix_s.eye
```

Comment:
One last thing is that, with the signature constraint, now these extra matrix/linalg functions in ndarray could only be defined in `owl_base_dense_ndarray.ml`. I think it's actually good to be specific about which functions are aliases from other modules. Anyway, thanks for this really good PR!

Reply:
Yea I agree it’s best not to alias the functions like this. The same thing is also done in owl_dense_ndarray.ml   

This is done mainly for the benefit of algodiff. But I think there might be a better way which does not involve injecting linalg functions into ndarray

Reply:
I can try and cook up something in a different PR